src/WebJobs.Script.WebHost/Diagnostics/SystemLogger.cs

```diff
@@ -173,12 +189,29 @@ public void Log<TState>(LogLevel logLevel, EventId eventId, TState state, Except
                     functionName = string.IsNullOrEmpty(fex.MethodName) ? string.Empty : fex.MethodName.Replace("Host.Functions.", string.Empty);
                 }
 
-                (innerExceptionType, innerExceptionMessage, details) = exception.GetExceptionDetails();
-                formattedMessage = Sanitizer.Sanitize(formattedMessage);
-                innerExceptionMessage = innerExceptionMessage ?? string.Empty;
+                var exceptionDetails = exception.GetExceptionDetails();
+
+                if (_logRpcExceptionDetails || !exceptionDetails.ExceptionType.Equals(_rpcExceptionName, StringComparison.Ordinal))
```

Comment:
Can we just check the exception type? The type should be accessible to this class.

```suggestion
                if (_logRpcExceptionDetails || exception is not RpcException)
```

Reply:
We determine the type of the innermost exception and use that for comparison. If the innerException is null, we default to `exception`.

Reply:
Can we do the same logic here then, but comparing by type and not string?